components/org.wso2.carbon.identity.application.authenticator.samlsso/src/test/java/org/wso2/carbon/identity/application/authenticator/samlsso/util/SSOUtilsTest.java

```diff
@@ -0,0 +1,185 @@
+import org.wso2.carbon.identity.application.authenticator.samlsso.exception.SAMLSSOException;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Unit test cases for SSOUtils
+ */
+public class SSOUtilsTest {
+
+    @Test
+    public void testCreateID() {
+
+        Assert.assertNotNull(SSOUtils.createID(), "Failed to generate an ID");
+    }
+
+    @Test
+    public void testEncode() {
+
+        Assert.assertEquals(SSOUtils.encode(TestConstants.XML_STRING), TestConstants.ENCODED_STRING);
+    }
+
+    @Test
+    public void testDecode() throws SAMLSSOException {
+
+        Assert.assertEquals(SSOUtils.decode(TestConstants.REDIRECT_REQUEST), TestConstants.REDIRECT_DECODED_STRING);
+    }
+
+    @Test (expectedExceptions = RuntimeException.class)
+    public void testDecodeEmptyRequest() throws SAMLSSOException {
+
+        String emptyRedirectRequest = "";
+        SSOUtils.decode(emptyRedirectRequest);
+    }
+
+    @Test(expectedExceptions = {SAMLSSOException.class})
+    public void testDecodeForInvalidRequest() throws SAMLSSOException {
+
+        SSOUtils.decode(TestConstants.INVALID_REDIRECT_REQUEST);
+    }
+
+    @Test
+    public void testDecodeForPost() throws SAMLSSOException {
+
+        Assert.assertEquals(SSOUtils.decodeForPost(TestConstants.POST_REQUEST), TestConstants.POST_DECODED_STRING);
+    }
+
+    @Test
+    public void testIsAuthnRequestSigned() {
+
+        Assert.assertEquals(false, SSOUtils.isAuthnRequestSigned(null));
+
+        Map<String, String> properties = new HashMap<>();
+        Assert.assertEquals(false, SSOUtils.isAuthnRequestSigned(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_AUTHN_REQ_SIGNED, "false");
+        Assert.assertEquals(false, SSOUtils.isAuthnRequestSigned(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_AUTHN_REQ_SIGNED, "true");
+        Assert.assertEquals(true, SSOUtils.isAuthnRequestSigned(properties));
+    }
+
+    @Test
+    public void testIsLogoutEnabled() {
+
+        Assert.assertEquals(false, SSOUtils.isLogoutEnabled(null));
+
+        Map<String, String> properties = new HashMap<>();
+        Assert.assertEquals(false, SSOUtils.isLogoutEnabled(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_ENABLED, "false");
+        Assert.assertEquals(false, SSOUtils.isLogoutEnabled(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_ENABLED, "true");
+        Assert.assertEquals(true, SSOUtils.isLogoutEnabled(properties));
+    }
+
+    @Test
+    public void testIsLogoutRequestSigned() {
+
+        Assert.assertEquals(false, SSOUtils.isLogoutRequestSigned(null));
+
+        Map<String, String> properties = new HashMap<>();
```

Comment:
Again isn't this writing multiple tests in a single test method ? What if the first one fails ? Second one will at least not execute.